DTS-STN/canadian-dental-care-plan
# Refactor 404 not found pages and routes

### Description

While working on [AB#2970](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/2970), I discovered a problem with how `404` statuses were being handled in our error boundaries.

If a loader or action returned a `404` response, the error boundary would render the server error component ("something went wrong!") instead of the 404 component.

I tried to fix it by just rendering the 404 component from `$.tsx` directly from the error boundary, but this was causing all kinds of layout rendering issues. Ultimately, I had to refactor both `$.tsx` and the error boundaries.

It now works when you hit a missing route (ie: `/asdf`), and also if a loader or action returns a `404` status code.

### Checklist


### Test Instructions

- Navigate to <http://localhost:3000/galneryus> and verify the 404 page works
- **(optional)** Hack one of the loaders to return `new Response(null, { status: 404 })` and verify the rendered not found page works
